src/clr-angular/forms/select/providers/option-selection.service.spec.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {ElementRef} from "@angular/core";
+import {Subscription} from "rxjs";
+
+import {IfOpenService} from "../../../utils/conditional/if-open.service";
+import {ClrOption} from "../option";
+
+import {OptionSelectionService} from "./option-selection.service";
+
+export default function() {
+    describe("Option Selection Service", () => {
+        let optionSelectionService: OptionSelectionService;
+        let fakeOption1: ClrOption;
+        let fakeOption2: ClrOption;
+
+        beforeEach(() => {
+            optionSelectionService = new OptionSelectionService();
+            fakeOption1 = new ClrOption(new IfOpenService(), new ElementRef(null), null, optionSelectionService);
+            fakeOption2 = new ClrOption(new IfOpenService(), new ElementRef(null), null, optionSelectionService);
+        });
+
+        it("updates the selection with the selected option", () => {
+            let selectedOption: ClrOption;
+            const subscription: Subscription =
+                optionSelectionService.selectionChanged.subscribe((option: ClrOption) => {
+                    selectedOption = option;
+                });
+
+            optionSelectionService.updateSelection(fakeOption1);
+
+            expect(selectedOption).toBe(fakeOption1);
+
+            subscription.unsubscribe();
+        });
+
+        it("replaces the current selection with the new selection", () => {
+            let selectedOption: ClrOption;
+            const subscription: Subscription =
+                optionSelectionService.selectionChanged.subscribe((option: ClrOption) => {
+                    selectedOption = option;
+                });
+
+            optionSelectionService.updateSelection(fakeOption1);
+
+            expect(selectedOption).toBe(fakeOption1);
+
+            optionSelectionService.updateSelection(fakeOption2);
+
+            expect(selectedOption).toBe(fakeOption2);
+
+            subscription.unsubscribe();
+        });
+
+        it("updates the selected flag of the option when it is selected", () => {
+            expect(fakeOption1.selected).toBe(false);
+            expect(fakeOption2.selected).toBe(false);
+
+            optionSelectionService.updateSelection(fakeOption1);
+
+            expect(fakeOption1.selected).toBe(true);
+            expect(fakeOption2.selected).toBe(false);
+
+            optionSelectionService.updateSelection(fakeOption2);
+
+            expect(fakeOption1.selected).toBe(false);
+            expect(fakeOption2.selected).toBe(true);
+        });
+
+        it("provides an observable to subscribe to change in option selection", () => {
+            expect(optionSelectionService.selectionChanged).toBeDefined();
+        });
+
+        it("notifies when a selection has been updated", () => {
```

Comment:
I'm worried that all of your tests break if this one breaks. Not sure if you can improve on that, but all the tests above already subscribe to this observable and rely on it to check their own stuff. It's okay if you don't change anything, I'm just voicing mild concerns.

Reply:
Sorry but I didn't understand the concern that you had here. The tests in this file have been updated to test the new pattern. Please let me know if you still feel that there is an issue here.